### DISCUSSION TITLE
Risk of Apple certificate revocation

### DISCUSSION THREAD

**Discussion by chimp1984** [bisq-network/bisq#6341] [General]
According to https://eclecticlight.co/2020/08/09/last-week-on-my-mac-heavy-hand-on-the-kill-switch/ when Apple decides to revoke the code signing certificate it 

> "tells you that the app which you’ve been using and trusting will damage your computer, and should be moved to the Trash". 

I am not sure how Windows deals with it, but i would be surprised if they are not worse (deriving from their track record).

I think the code signing we use for OSX and Windows to avoid the annoying security warnings are a high risk that those 2 companies could shut down a huge percentage of our user base and leave a huge reputation damage as many ppl might think Bisq was detected as malware.

The downside to remove the code signing is that the users of those OS need to do some extra steps (which unforunately change with each new OS update at OSX). I guess most Bitcoin related apps do not do code signing anyway and user know already how to deal with it. We can offer "how-to" instructions at the release download page and on the wiki to minimize the pain.
Another downside is that the code signing (if not abused for political reasons) brings some level of extra security. But if people download via the Bisq app or do manually the pgp sig check they get basically the same security.
  
I would recommend to remove the code signing to avoid that risk. Apple had a bad track record when they kicked out all Bitcoin related apps back in 2013 from the app store. We should not trust them to not repeat that behaviour in future.

**Comment by chrisBonGit** [comment]
Attempting to install Bisq v1.9.6 on macOS v12.6.1
Signature check and hash check on jar file after install both passed.
MacOS refuses to run, stating that Bisq is damaged. Hence I can't use the force run anyway fix since system general doesn't allow me to.
I'll try a reboot to see if this makes the OS forget that it thinks the executable is damaged...

**Comment by ghost** [comment]
MacOS instructions here: https://bisq.wiki/Downloading_and_installing#macOS

"**rex4539**" on Matrix suggested a simpler solution on mac : right click and select Open the first time to bypass the quarantine issue.
Would be good to know if that solution works.

**Reply by chrisBonGit** [reply]
Thanks for the ideas.

Re "rex4539"'s idea: I gave it a go, but I get the same error message, 'Bisq is damaged and can't be opened'.

I've followed the MacOS instructions, but because the error is a statement that the file is damanged, I am not able to force open/execution by bypassing in the system/security&Privacy options.

(FWIW I have managed to install and execute bisq on a different device, running windows 10.)

Might this problem be because I'm using a M1 mac mini (2020), i.e. is the image in the .dmg file only executable on the intel chips rather than the apple silicon hardware?

**Reply by n-scope** [reply]
Nope I am running an intel MacOS BIgSur and get the same result. There is something wrong with the binary.

**Comment by brn004** [comment]
Can we add bisq to macOS homebrew?

**Reply by freezoom1** [reply]
I was able to install w/ the hombrew cask install to the latest version v1.9.14

`brew install bisq`

Note - I had the same problems w/ the MacOS bisq.dmg install ("send to trash") and was able to overcome this with the brew install command instead. This did install properly on MacOS Monterey 12.7.2

**Comment by n-scope** [comment]
MacOS BigSur, intel: 'Bisq is damaged and can't be opened'.
Version 1.9.5 runs properly.

**Reply by ripcurlx** [reply]
Did you try already the workaround mentioned in the wiki? https://bisq.wiki/Downloading_and_installing#macOS

**Reply by chrisBonGit** [reply]
Update: The link shared by ripcurlx has instructions for users of macOS Verntura (13+), specifically there's an instruction to run the following in a terminal 'sudo xattr -rd com.apple.quarantine /Applications/Bisq.app'
I am still on maxOS Monterey BUT this fix also worked for me. So, if you are struggling I advise you to try the above (even if you've not yet upgraded to Ventura)

**Reply by thibistaken** [reply]
This solution worked for me as well. I'm on MacOS Monterey 12.6 running Bisq 1.9.9.

**Reply by david-worrall** [reply]
This solution also works for me: MacOS Catalina 10.15.7 running Bisq 1.9.9.

**Reply by bbpoizon** [reply]
is there an updated workaround? I'm on 15.4.1 and this no longer works

**Reply by rex4539** [reply]
Problem is `option -r not recognized`.

Use `xattr -d com.apple.quarantine /Applications/Bisq.app`
